src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService_Helpers.cs

```diff
@@ -0,0 +1,178 @@
+                inheritanceRelationshipWithOriginalMember,
+                definition);
+        }
+
+        private static async Task<InheritanceMemberItem> CreateInheritanceMemberInfoForMemberAsync(
+            Solution solution,
+            ISymbol memberSymbol,
+            int lineNumber,
+            ImmutableArray<ISymbol> implementingMembers,
+            ImmutableArray<ISymbol> implementedMembers,
+            ImmutableArray<ISymbol> overridenMembers,
+            ImmutableArray<ISymbol> overridingMembers,
+            CancellationToken cancellationToken)
+        {
+            var implementingMemberItems = await implementingMembers
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Implementing, cancellationToken)).ConfigureAwait(false);
+            var implementedMemberItems = await implementedMembers
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Implemented, cancellationToken)).ConfigureAwait(false);
+            var overridenMemberItems = await overridenMembers
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Overridden, cancellationToken)).ConfigureAwait(false);
+            var overridingMemberItems = await overridingMembers
+                .SelectAsArrayAsync(symbol => CreateInheritanceItemAsync(solution, symbol, InheritanceRelationship.Overriding, cancellationToken)).ConfigureAwait(false);
+
+            return new InheritanceMemberItem(
+                lineNumber,
+                memberSymbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                memberSymbol.GetGlyph(),
+                implementingMemberItems.Concat(implementedMemberItems)
+                    .Concat(overridenMemberItems)
+                    .Concat(overridingMemberItems));
+        }
+
+        /// <summary>
+        /// For the <param name="memberSymbol"/>, get all the implemented symbols.
+        /// Table for the mapping between images and inheritanceRelationship
+        /// Implemented : I↓
+        /// Implementing : I↑
+        /// Overridden: O↓
+        /// Overriding: O↑
+        /// </summary>
+        private static async Task<ImmutableArray<ISymbol>> GetImplementedSymbolsAsync(
+            Solution solution,
+            ISymbol memberSymbol,
+            CancellationToken cancellationToken)
+        {
+            if (memberSymbol is INamedTypeSymbol { IsSealed: false } namedTypeSymbol)
+            {
+                var derivedTypes = await GetDerivedTypesAndImplementationsAsync(solution, namedTypeSymbol, cancellationToken).ConfigureAwait(false);
+                return derivedTypes.CastArray<ISymbol>().ToImmutableArray();
+            }
+
+            if (memberSymbol is IMethodSymbol or IEventSymbol or IPropertySymbol
+                 && memberSymbol.ContainingSymbol.IsInterfaceType())
```

Comment:
so this finds implementations of interface members.  where do you do the reverse?  i.e. given a class member, how do you find/add the interface members it implements?

Reply:
```
     // Go up the inheritance chain to find all the implemented targets.
     var implementingMembers = memberSymbol.ExplicitOrImplicitInterfaceImplementations();
```
Link is here: https://github.com/Cosifne/roslyn/blob/30ac4cf331fd4cd247908c512842720d4acb205d/src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService.cs#L142